lldb/tools/debugserver/source/RNBRemote.cpp

```diff
@@ -3944,15 +3955,22 @@ rnb_err_t RNBRemote::HandlePacket_v(const char *p) {
         // The order of these checks is important.  
         if (process_does_not_exist (pid_attaching_to)) {
           DNBLogError("Tried to attach to pid that doesn't exist");
-          std::string return_message = "E96;";
-          return_message += cstring_to_asciihex_string("no such process.");
+          std::string return_message = "E96";
```

Comment:
I see this pattern in lots of places, is it worth abstracting out? Something like:

```
std::string CreateAttachError(const char *additional_context = nullptr) {
    std::string message = "E96";
    if (additional_context)
        message += ";" + cstring_to_asciihex_string(additional_context);
    return message;
}
```

What do you think?

Reply:
Then this line becomes:
```
std::string return_message = CreateEXXPacket("E96", "no such process");
```

Reply:
I'm thinking of having a SendErrorPacket() instead of SendPacket() with an error string optional argument, and having all error packets route through that, and having all SendPacket("Exx")'s go through that instead.  @bulbazord @clayborg what do you think?  I'm having trouble deciding if I want to have a method that constructs the error string and the caller passes it to SendPacket or if I want to have a method to do both.

Reply:
> I'm thinking of having a SendErrorPacket() instead of SendPacket() with an error string optional argument, and having all error packets route through that, and having all SendPacket("Exx")'s go through that instead. @bulbazord @clayborg what do you think? I'm having trouble deciding if I want to have a method that constructs the error string and the caller passes it to SendPacket or if I want to have a method to do both.

I like that idea!

Reply:
Sounds great to me 😄